File: app/src/main/java/net/squanchy/support/injection/CurrentTimeModule.java
package net.squanchy.support.injection;

import net.squanchy.support.system.CurrentTime;

import dagger.Module;
import dagger.Provides;

@Module
public class CurrentTimeModule {

    @Provides
    CurrentTime provideCurrentTime() {
        return new CurrentTime();
    }
}

File: app/src/debug/java/net/squanchy/support/checksum/CurrentTimeModule.kt
package net.squanchy.support.checksum

import dagger.Module
import dagger.Provides
import net.squanchy.support.system.CurrentTime
import net.squanchy.support.system.DebugCurrentTime

@Module
class CurrentTimeModule {

    @Provides
    internal fun provideCurrentTime(): CurrentTime = DebugCurrentTime()
}
